### ISSUE TITLE
Quiver 3.0.1 Beta

### ISSUE THREAD

**Issue by ylian** [HappenApps/Quiver#361]
Quiver 3.0.1 beta comes with the following changes:
- Swipe to change the main window layout
- Copy a note as Markdown (https://github.com/HappenApps/Quiver/issues/251)
- Copy a note as HTML source (https://github.com/HappenApps/Quiver/issues/243)
- Copy a note as Code Only (https://github.com/HappenApps/Quiver/issues/338)
- Email a note as PDF, Markdown, or Plain Text
- Fixed an issue that the Evernote importer fails silently if the ENEX file contains empty resources (https://github.com/HappenApps/Quiver/issues/253)
- Added support for a custom URL scheme: quiver:///notes/{noteId} (https://github.com/HappenApps/Quiver/issues/294, https://github.com/HappenApps/Quiver/issues/97)
- Fixed rendering issues with fenced code blocks in Markdown cells (https://github.com/HappenApps/Quiver/issues/249)
- Allow importing .txt files as Markdown
- Restored support for using `\[ \]`, `\( \)` as LaTeX delimiters
- Option to config LaTeX delimiters (https://github.com/HappenApps/Quiver/issues/318)
- Option to allow manual escaping of special Markdown characters in LaTeX (compatibility with Hexo)
- ⌃⌘P now hides the floating preview window when it's already visible
- Fixed a bug that Cmd+Backspace didn't work in search fields. (https://github.com/HappenApps/Quiver/issues/303)
- Fixed a bug that pressing the escape key didn’t clear the search field under the note list (https://github.com/HappenApps/Quiver/issues/308)
- After deleting or moving a note, the note list no longer jumps to the top. (https://github.com/HappenApps/Quiver/issues/293)
- Fixed an issue that when exporting a note as Markdown, image URLs were not properly converted. (https://github.com/HappenApps/Quiver/issues/351)
- Added a menu item to edit note tags (https://github.com/HappenApps/Quiver/issues/348)
- Added a cancel button to the file attachment dialog (https://github.com/HappenApps/Quiver/issues/244)

I'm still debugging #309, #349. Please test the build in the meantime. Thanks.

https://dl.dropboxusercontent.com/u/3445731/Quiver%203.0.1%20Build%2073.zip

**Comment by JMichaelTX** [comment]
> Quiver 3.0.1 beta comes with the following changes:

Outstanding!  :+1:

**Comment by paulelms** [comment]
👍🏼

**Comment by paulelms** [comment]
Request for applescript automation possibilities: we need to gain access to a certain type of cells in the current note, edit them, add new cells, etc.
Thanks!

**Comment by elbie-x** [comment]
Re: [#338](https://github.com/HappenApps/Quiver/issues/338), Copy As Code Only:

It’s great that you've implemented it so that _code **blocks**_ _inside Markdown cells_ **don’t** get copied out. That way if I have multiple kinds of code in a note, but only one that I’m actually going to copy and paste to run in another program, I don’t get bad-syntax, irrelevant chunks that are actually in another language and just comments to myself. 

In short: Copy As Code Only == Copy Only Code _Cells_, which is perfect. Thank you!

**Comment by marlowe310** [comment]
Hopefully this isn't just user error -- but it seems like the "email as" functions in the new beta (markdown, plain text, PDF) only work when Apple Mail is set as the default mail handler. When I changed it to Airmail 2 (my daily driver), it no longer creates the email/attaches the file. Other "mail-to" links are working, so I suspect the issue is limited to Quiver. Thanks so much for all the hard work!

**Comment by ylian** [comment]
@marlowe310 That's probably expected. I'm not using "mail-to" links, since "mail-to" links don't support attachments, and have trouble with long contents. I'm using the `NSSharingService` which probably only works with Mail, but I will check if it's possible to fix.

**Comment by ylian** [comment]
@marlowe310 Can you check if the Share menu in Safari works if you set the default mail client to Airmail?

**Comment by ylian** [comment]
@marlowe310 Just tried. Obviously if Mail is not the default mail client, it doesn't even show up in the Share menu. Airmail provides its own Share extension. I'm checking if it's possible to use that.

**Comment by marlowe310** [comment]
Hey just tested it (before seeing your own results), and I had the same result (as expected, it seems). Thanks so much for going out of your way to test it out — let me know if there’s anything more I can do to help!

On January 21, 2016 at 5:54:30 PM, Yaogang Lian (notifications@github.com) wrote:

@marlowe310 Just tried. Obviously if Mail is not the default mail client, it doesn't even show up in the Share menu. Airmail provides its own Share extension. I'm checking if it's possible to use that.

**Comment by JMichaelTX** [comment]
@ylian:  I hope whatever system you are using will work with Outlook 2011, my only email client.  A number of other apps recognize and use Outlook, like the browsers and Adobe Acrobat.

**Comment by ylian** [comment]
@JMichaelTX Just tried Outlook 2011. It doesn't come with a Share extension, so I have no way to make it work. Other apps probably use `mailto:` links, but that may cut off long emails, and doesn't support attachment.

**Comment by ylian** [comment]
Quiver 3.0.1 Build 73: https://dl.dropboxusercontent.com/u/3445731/Quiver%203.0.1%20Build%2073.zip

Changes:
- Improved swipe gesture detection
- Fixed an issue that the date label in the note list may be cutoff (https://github.com/HappenApps/Quiver/issues/252)
- When emailing a note, if Apple Mail is not the default mail client, look for Airmail.
- Cmd+click to open a note link in a separate window

**Comment by marlowe310** [comment]
Thanks VERY much!

On January 21, 2016 at 8:24:27 PM, Yaogang Lian (notifications@github.com) wrote:

Quiver 3.0.1 Build 73: https://dl.dropboxusercontent.com/u/3445731/Quiver%203.0.1%20Build%2073.zip

Changes:

Improved swipe gesture detection
Fixed an issue that the date label in the note list may be cutoff (#252)
When emailing a note, if Apple Mail is not the default mail client, look for Airmail.
Cmd+click to open a note link in a separate window

**Comment by ylian** [comment]
@marlowe310 No problem. Note that you need to enable Airmail's Compose extension for this to work.

**Comment by JMichaelTX** [comment]
> Just tried Outlook 2011. It doesn't come with a Share extension, so I have no way to make it work

That's too bad.  But thanks very much for checking.  If you can provide good AppleScript support I can probably write a script to do this.

**Comment by ylian** [comment]
3.0.1 is released.
